angular_token_auth.js

```diff
@@ -348,15 +348,22 @@
                 }).success(function (data) {
                     authFactory.setAuth(data);
                     deferred.resolve(data);
-                }).error(function (data) {
-                    deferred.reject(data);
-                });
+                }).error(deferred.reject);
 
                 return deferred.promise;
             },
             logout: function () {
-                authFactory.clearAuth();
-                $location.url(MODULE_SETTINGS.LOGOUT_REDIRECT_URL);
+                var deferred = $q.defer();
+
+                $http.delete(PROJECT_SETTINGS.API_ROOT + MODULE_SETTINGS.ENDPOINT)
+                    .success(function (data) {
+                        authFactory.clearAuth();
+                        $location.url(MODULE_SETTINGS.LOGOUT_REDIRECT_URL);
+                        deferred.resolve(data);
+                    })
+                    .error(deferred.reject);
+
+                return deferred.promise;
```

Comment:
This feels like a breaking change because current code using this library won't expect a failing http request to not log you out.

Perhaps move the `authFactory.clearAuth` and `$location.url` lines into an `always` or `finally` so that they always run, regardless of the success of the session delete http request?